tianshou/env/worker/base.py

```diff
@@ -0,0 +1,71 @@
+import gym
+import numpy as np
+from abc import ABC, abstractmethod
+from typing import List, Tuple, Optional, Callable, Any
+
+
+class EnvWorker(ABC, gym.Env):
```

Comment:
Not a pool. A worker here only interacts with one environment and the `VectorEnv` class is the pool for handling many workers.

Reply:
Yes I noticed that after writing my comment and I thought it was better to get it as is, just for you to know I was confused in the first place x) Meaning that some comments are missing in the code, the implementation is not self explanatory, or I'm just sleepy or slow minded :)

Reply:
Yes, it is somewhat confusing at first glance. Currently there is a comment saying this is ``An abstract worker for an environment.``, maybe we should emphasis that this is a worker for **an** environment.

Reply:
No at worker level it is fine (or at least, fine enough). I rather got confused to VecEnv. The information that is missing is that VecEnv is not a usual pool, in a sense that it does not really handle parallelism by itself. Being asynchronous or not is to the resort of the sole worker and at its discretion, with no explicit mechanism for the pool to know this information in advance (hence the unusual `wait` mechanism). And this is what is actually confusing, since at first glance vecenv looks like a vector of pool more than a pool itself (and with the current implement, it totally could be) !